Description: Fix hyphen rendering in option tags
 The docbook2man script incorrectly renders <option> tags using a standard
 hyphen (-) instead of the roff escape sequence for a minus sign (\-).
 This causes flags to fail when copied and pasted from man pages into
 a terminal, and triggers the Lintian tag 'hyphen-used-as-minus-sign'.
 .
 This patch modifies the SGML handler to capture the option content
 and escape hyphens before outputting.
Author: YIN Renlong
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/docbook-utils/+bug/19987
Bug-Debian: https://bugs.debian.org/208967
Last-Update: 2026-01-21

--- a/helpers/docbook2man-spec.pl
+++ b/helpers/docbook2man-spec.pl
@@ -519,8 +519,12 @@
 sgml('<GROUP>', \&group_start);
 sgml('</GROUP>', \&group_end);
 
-sgml('<OPTION>', \&bold_on);
-sgml('</OPTION>', \&font_off);
+sgml('<OPTION>', sub {
+	&bold_on;
+	push_output('string');
+});
+sgml('</OPTION>', sub {
+	my $content = pop_output();
+	$content =~ s/-/\\-/g;
+	output $content;
+	&font_off;
+});
 
 man_sgml('<SBR>', "\n    ");
